File: backend/app/domain/case.py
"""Case domain enumerations."""

from enum import StrEnum


class CaseStatus(StrEnum):
    """Controlled lifecycle states for an investigation case."""

    OPEN = "OPEN"
    IN_PROGRESS = "IN_PROGRESS"
    ON_HOLD = "ON_HOLD"
    COMPLETED = "COMPLETED"
    ARCHIVED = "ARCHIVED"


class CasePriority(StrEnum):
    """Controlled priority levels for an investigation case."""

    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"

File: frontend/src/types/case.ts
export type CaseStatus =
  | "OPEN"
  | "IN_PROGRESS"
  | "ON_HOLD"
  | "COMPLETED"
  | "ARCHIVED";

export type CasePriority = "LOW" | "MEDIUM" | "HIGH" | "CRITICAL";

export interface CaseRecord {
  id: string;
  case_number: string;
  title: string;
  description: string | null;
  status: CaseStatus;
  priority: CasePriority;
  created_at: string;
  updated_at: string;
}

export interface CaseListData {
  items: CaseRecord[];
  total: number;
  limit: number;
  offset: number;
}
